### ISSUE TITLE
Compilation warnings - The exported identifier is not declared in Babel's scope tracker

### ISSUE THREAD

**Issue by andrewcourtice** [dosentmatter/babel-plugin-const-enum#20]
Hi

Thanks for creating this plugin. It is really helpful!

When compiling I receive the following warning for every instance of a `const enum`:
```
The exported identifier "STATE" is not declared in Babel's scope tracker
as a JavaScript value binding, and "@babel/plugin-transform-typescript"
never encountered it as a TypeScript type declaration.
It will be treated as a JavaScript value.

This problem is likely caused by another plugin injecting
"STATE" without registering it in the scope tracker. If you are the author
 of that plugin, please use "scope.registerDeclaration(declarationPath)".
The exported identifier "STATE" is not declared in Babel's scope tracker
as a JavaScript value binding, and "@babel/plugin-transform-typescript"
never encountered it as a TypeScript type declaration.
It will be treated as a JavaScript value.
```

Here is my babel config file:
```
module.exports = {
    presets: [
        ['@babel/env', {
            useBuiltIns: 'entry',
            corejs: 3
        }],
        '@babel/typescript',
    ],
    plugins: [
        ['const-enum', {
            transform: 'constObject'
        }],
        '@babel/transform-typescript',
        '@babel/plugin-syntax-dynamic-import'
    ]
};
```

Is this a bug or am I missing something in the configuration?

Thanks

**Comment by dosentmatter** [comment]
@andrewcourtice
> Thanks for creating this plugin. It is really helpful!

You're welcome! Thanks for using it and thanks for reporting this issue.

Your babel config is correct, but it can be simplified.
You shouldn't need `@babel/plugin-transform-typescript` because it is already included in `@babel/preset-typescript`.
You shouldn't need `@babel/plugin-syntax-dynamic-import` because it is already included in `@babel/preset-env`

```js
// .babelrc.js
module.exports = {
  plugins: [
    [
      "const-enum",
      {
        transform: "constObject"
      }
    ],
  ],
  presets: [
    [
      "@babel/env",
      {
        useBuiltIns: "entry",
        corejs: 3
      }
    ],
    "@babel/typescript"
  ],
};
```

From what I can guess, you have an enum named `STATE` and you are exporting it?
I found the [PR in `@babel/plugin-transform-typescript`](https://github.com/babel/babel/pull/10174) that fixes this issue, however I still might want to call the `scope.registerDeclaration(declarationPath)` to be more correct.

That PR fixes an issue where exports that aren't registered in the babel scope are assumed to be TS types and are removed. This may happen when plugins declare a new variable without registering them into the scope. However, I am not declaring anything new with the `constObject`. I just reuse the enum name. I'm not sure if I need to register the `constObject` into scope or if it is already done.

Can you tell me what package versions you are on for:
1. `babel-plugin-const-enum`
2. `@babel/preset-env`
3. `@babel/typescript`

I'm still unable to reproduce your issue. If you have a minimally reproducible example or repo, that would be helpful.

**Comment by dosentmatter** [comment]
I think I found out how to reproduce:

```sh
$ cat .babelrc.js 
module.exports = {
  plugins: [
    [
      "const-enum",
      {
        transform: "constObject"
      }
    ]
  ],
  presets: [
    [
      "@babel/env",
      {
        useBuiltIns: "entry",
        corejs: 3
      }
    ],
    "@babel/typescript"
  ]
};
```

```sh
$ cat <<EOF | npx babel -f stdin.ts
const enum MyEnum {
  A = 1,
  B = A * 2,
  C,
  D = C,
  E = D ** 2,
  F,
  G = F * E,
  H,
  I = H << 20
}
export { MyEnum }
EOF
The exported identifier "MyEnum" is not declared in Babel's scope tracker
as a JavaScript value binding, and "@babel/plugin-transform-typescript"
never encountered it as a TypeScript type declaration.
It will be treated as a JavaScript value.

This problem is likely caused by another plugin injecting
"MyEnum" without registering it in the scope tracker. If you are the author
 of that plugin, please use "scope.registerDeclaration(declarationPath)".
The exported identifier "MyEnum" is not declared in Babel's scope tracker
as a JavaScript value binding, and "@babel/plugin-transform-typescript"
never encountered it as a TypeScript type declaration.
It will be treated as a JavaScript value.

This problem is likely caused by another plugin injecting
"MyEnum" without registering it in the scope tracker. If you are the author
 of that plugin, please use "scope.registerDeclaration(declarationPath)".
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MyEnum = void 0;
const MyEnum = {
  A: 1,
  B: 2,
  C: 3,
  D: 3,
  E: 9,
  F: 10,
  G: 90,
  H: 91,
  I: 95420416
};
exports.MyEnum = MyEnum;
```

I will look at this more in my free time. I think I know how to fix it.

**Comment by andrewcourtice** [comment]
Thanks @dosentmatter. I really appreciate you looking into this.

In answer to your previous question, here are the versions of the various aforementioned packages:

```json
"devDependencies": {
    "@babel/core": "^7.8.4",
    "@babel/plugin-syntax-dynamic-import": "^7.8.3",
    "@babel/plugin-transform-typescript": "^7.7.4",
    "@babel/preset-env": "^7.8.4",
    "@babel/preset-typescript": "^7.7.4",
    "babel-plugin-const-enum": "^0.0.5",
}
```

And this is how I am declaring my `enums` (if it helps):

```typescript
export const enum PERIOD_TYPE {
    month = 'MONTH',
    quarter = 'QUARTER',
    year = 'YEAR'
};

export default PERIOD_TYPE;
```

Let me know if there is anything else I can do to help.

Thanks again.

**Comment by dosentmatter** [comment]
@andrewcourtice, you're welcome. Thanks for providing me the information.

I made a fix and published to version `1.0.1`. Try it out and let me know if you still have problems.

**Comment by andrewcourtice** [comment]
@dosentmatter You sir, are a legend! The update worked perfectly. 

Thanks for sorting that out so quickly 👍Much appreciated.
